src/map/script.c

```diff
@@ -18030,10 +18030,12 @@ static BUILDIN(npcshopdelitem)
 		unsigned int nameid = script_getnum(st,i);
 
 		ARR_FIND(0, size, n, nd->u.shop.shop_item[n].nameid == nameid);
-		if (n < size) {
-			memmove(&nd->u.shop.shop_item[n], &nd->u.shop.shop_item[n+1], sizeof(nd->u.shop.shop_item[0])*(size-n));
-			size--;
+		if (n == size) {
+			continue;
+		} else if (n < size) {
```

Comment:
I think need change to ``} else if (n < size + 1) {``

Reply:
I think it should be change to (n < size - 1)

Reply:
yes, ssory should be -1

Reply:
Fixed